framework/doc/content/source/meshmodifiers/ElementSubdomainModifier.md

```diff
@@ -0,0 +1,99 @@
+# ElementSubdomainModifier
+
+## Overview
+
+The `ElementSubdomainModifier` modifies an element subdomain ID. This is inherited by other mesh modifiers, such as [CoupledVarThresholdElementSubdomainModifier.md] and [TimedSubdomainModifier.md], which implement different criteria for a subdomain ID to be modified.
+
+The `ElementSubdomainModifier` can model
+
+- Element death (with applications in ablation, fracture, etc.);
+- Element activation (with applications in additive manufacturing, sintering, solidification, etc.);
+- Moving interface (with applications in metal oxidation, phase transformation, melt pool, etc.).
+
+The `ElementSubdomainModifier` only changes the element's subdomain. It inherits from `ElementSubdomainModifierBase`, which handles the corresponding
+
+- Moving boundary/interface nodeset/sideset modification,
+- Solution reinitialization, and
+- Stateful material property reinitialization,
+
+all of which are demonstrated below.
+
+Consider a unit square domain decomposed by a vertical line $x=0.25$. The elements on the left side of the vertical line have subdomain ID of 1, and the elements on the right side have subdomain ID of 2. The `ElementSubdomainModifier` is used to change the subdomain ID from 2 to 1 for elements within a circle of radius 0.5 whose center moves along the bottom side towards the right.
+
+## Moving boundary/interface nodeset/sideset modification
+
+!alert note
+If the moving boundary is defined completely around a subdomain, or between subdomains, then [SidesetAroundSubdomainUpdater.md] may be more useful to use in conjunction with the `ElementSubdomainModifier`, rather than using the `moving_boundaries` and `moving_boundary_subdomain_pairs` parameters in `ElementSubdomainModifier`.
+
+The change of element subdomains will alter the definitions of certain sidesets and nodesets. The parameters `moving_boundaries` and `moving_boundary_subdomain_pairs` can optionally be used to modify the corresponding sidesets/nodesets over the elements that change subdomain. The names of the boundaries are specified in `moving_boundaries`, and the pair of subdomains that each boundary in `moving_boundaries` lies between must be specified in the corresponding `moving_boundary_subdomain_pairs`. The element side from the first subdomain of the pair is added to the boundary.
```

Comment:
this ensures the param names stay in sync with the docs. no need to do it for every single instance of each parameter, but each parameter is a good idea
```suggestion
The change of element subdomains will alter the definitions of certain sidesets and nodesets. The parameters [!param](/MeshModifiers/CoupledVarThresholdElementSubdomainModifier/moving_boundaries) and `moving_boundary_subdomain_pairs` can optionally be used to modify the corresponding sidesets/nodesets over the elements that change subdomain. The names of the boundaries are specified in `moving_boundaries`, and the pair of subdomains that each boundary in `moving_boundaries` lies between must be specified in the corresponding `moving_boundary_subdomain_pairs`. The element side from the first subdomain of the pair is added to the boundary.
```